src/daemon/usb.c

```diff
@@ -286,21 +286,22 @@ static void* _setupusb(void* context){
     /// The reason is, if the open fails or not open has been done until now,
     /// that struct member is set to 0, not to -1 or other negative value.
     /// So all usage of this kb->handle must be something like \c "kb->handle - 1", as you can find it in the code.
-    if(os_inputopen(kb))
-        goto fail;
-    ///
-    /// - The next action is to create a separate thread, which gets as parameter kb and starts with os_inputmain().
-    /// The thread is immediately detached so that it can return its resource completely independently if it should terminate.
-    if(pthread_create(&kb->inputthread, 0, os_inputmain, kb))
-        goto fail;
-    pthread_detach(kb->inputthread);
-    ///
-    /// - The same happens with os_setupindicators(),
-    /// which initially initializes all LED variables in kb to off and then starts the _ledthread() thread
-    /// with kb as parameter and then detaches it.
-    /// Here again only the generation of the thread can fail.
-    if(os_setupindicators(kb))
-        goto fail;
+    if(!os_inputopen(kb)) {
+        /// Successfully opened an input device
+        ///
+        /// - The next action is to create a separate thread, which gets as parameter kb and starts with os_inputmain().
+        /// The thread is immediately detached so that it can return its resource completely independently if it should terminate.
+        if(pthread_create(&kb->inputthread, 0, os_inputmain, kb))
```

Comment:
The input thread needs to always start, as it handles all URB Interrupts.

Reply:
So the os_inputmain function isn't dependent on anything set inside os_inputopen?

Reply:
In a way it is, but only if HID (0x01 or 0x03) packets are received.
The mousepad does not send any of those, so it's safe.